prboom2/src/MUSIC/portmidiplayer.c

```diff
@@ -463,36 +404,44 @@ static dboolean is_sysex_reset (unsigned char *data)
 
 static void writesysex (unsigned long when, int etype, unsigned char *data, int len)
 {
-  // sysex code is untested
-  // it's possible to use an auto-resizing buffer here, but a malformed
-  // midi file could make it grow arbitrarily large (since it must grow
-  // until it hits an 0xf7 terminator)
+  // sysex messages in midi files (smf 1.0 pages 6-7):
+  // complete:        (F0 ... F7)
+  // multi-packet:    (F0 ...) + (F7 ...) + ... + (F7 ... F7)
+  // escape sequence: (F7 ...)
+
   if (len + sysexbufflen > SYSEX_BUFF_SIZE - 1)
   {
-    lprintf (LO_WARN, "portmidiplayer: ignoring large or malformed sysex message\n");
+    // ignore messages that are too long
     sysexbufflen = 0;
     return;
   }
+
+  if (etype == MIDI_EVENT_SYSEX_SPLIT && sysexbufflen == 0)
```

Comment:
Looks impressive! A few questions:
What happens if we send an escape sequence? Do you have .mid files with `MIDI_EVENT_SYSEX_SPLIT` events?

Reply:
According to the midi specs, they're used to pass  characters that are not supposed to be sent to the device and is an exploit in the protocol. I guess an example would be sneaking an 0xFF through which would reboot/reset the device but is is simply the meta event status byte in a midi file. Better to drop the hacks, imo.